app-modules/interaction/tests/Models/Scopes/InteractionConfidentialScopeTest.php

```diff
@@ -35,55 +35,92 @@
 */
 
 use AdvisingApp\Authorization\Enums\LicenseType;
-use AdvisingApp\Interaction\Filament\Resources\InteractionResource\Pages\ListInteractions;
 use AdvisingApp\Interaction\Models\Interaction;
+use AdvisingApp\Interaction\Models\Scopes\InteractionConfidentialScope;
 use AdvisingApp\Team\Models\Team;
 use App\Models\User;
 
 use function Pest\Laravel\actingAs;
-use function Pest\Livewire\livewire;
 use function Tests\asSuperAdmin;
 
-test('ListInteration with display data using the is_confidential field', function () {
+test('Interaction model has applied global scope', function () {
+    Interaction::factory()->create();
+
+    expect(Interaction::hasGlobalScope(InteractionConfidentialScope::class))->toBeTrue();
+});
+
+test('Interactions model with fetch data count for created user', function () {
     $user = User::factory()->licensed(LicenseType::cases())->create();
     $user->givePermissionTo('interaction.view-any');
 
-    $userWithoutAttach = User::factory()->licensed(LicenseType::cases())->create();
-    $userWithoutAttach->givePermissionTo('interaction.view-any');
+    actingAs($user);
+    Interaction::factory()->count(10)->create([
+        'is_confidential' => true,
+        'user_id' => $user,
+    ]);
+
+    Interaction::factory()->count(10)->create([
+        'is_confidential' => true,
+    ]);
 
+    Interaction::factory()->count(10)->create([
+        'is_confidential' => false,
+    ]);
+
+    expect(Interaction::query()->count())->toBe(20);
+});
+
+test('Interactions model with fetch data count for team user', function () {
     $teamUser = User::factory()->licensed(LicenseType::cases())->create();
     $teamUser->givePermissionTo('interaction.view-any');
 
     $team = Team::factory()->hasAttached($teamUser, [], 'users')->create();
 
-    $confidentialInteraction = Interaction::factory()->hasAttached($user, [], 'confidentialAccessUsers')->hasAttached($team, [], 'confidentialAccessTeams')->count(10)->create([
+    actingAs($teamUser);
+    Interaction::factory()->hasAttached($team, [], 'confidentialAccessTeams')->count(10)->create([
         'is_confidential' => true,
     ]);
 
-    $nonConfidentialInteraction = Interaction::factory()->count(10)->create([
+    Interaction::factory()->count(10)->create([
+        'is_confidential' => true,
+    ]);
+
+    Interaction::factory()->count(10)->create([
         'is_confidential' => false,
     ]);
 
-    $allInteractions = $confidentialInteraction->merge($nonConfidentialInteraction);
+    expect(Interaction::query()->count())->toBe(20);
```

Comment:
Would shouldn't just check count. We should also be checking the actual records that are returned to see if they are the records we expect.